pkg/datagatherer/versionchecker/versionchecker_test.go

```diff
@@ -0,0 +1,357 @@
+	if err != nil {
+		t.Fatalf("failed to load config: %+v", err)
+	}
+
+	dg, err := config.NewDataGatherer(context.Background())
+	if err != nil {
+		t.Fatalf("failed create new dg %s", err)
+	}
+
+	rawResults, err := dg.Fetch()
+	if err != nil {
+		t.Fatalf("failed fetch data %s", err)
+	}
+
+	resultsJSON, err := json.MarshalIndent(rawResults, "", "  ")
+	if err != nil {
+		t.Fatalf("failed to marshal data %s", err)
+	}
+
+	expectedResultsJSON := fmt.Sprintf(`[
+  {
+    "pod": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-6d94489854-zpzhr",
+        "namespace": "example",
+        "selfLink": "/api/v1/namespaces/example/pods/example-6d94489854-zpzhr",
+        "uid": "efff9dae-28ca-42c3-be70-970731c44f67",
+        "resourceVersion": "32023849",
+        "creationTimestamp": null,
+        "labels": {
+          "app": "example"
+        },
+        "ownerReferences": [
+          {
+            "apiVersion": "apps/v1",
+            "kind": "ReplicaSet",
+            "name": "example-6d94489854",
+            "uid": "bb6c0f31-0e28-4c28-a81d-91b8d7bfed33",
+            "controller": true,
+            "blockOwnerDeletion": true
+          }
+        ]
+      },
+      "spec": {
+        "containers": [
+          {
+            "name": "example",
+            "image": "%s/jetstack/example:v1.0.0",
+            "command": [
+              "sh",
+              "-c"
+            ],
+            "resources": {}
+          }
+        ]
+      },
+      "status": {
+        "containerStatuses": [
+          {
+            "name": "example",
+            "state": {},
+            "lastState": {},
+            "ready": false,
+            "restartCount": 0,
+            "image": "",
+            "imageID": "is set"
+          }
+        ]
+      }
+    },
+    "result": {
+      "CurrentVersion": "v1.0.0",
+      "LatestVersion": "v1.0.1",
+      "IsLatest": false,
+      "ImageURL": "%s/jetstack/example"
+    }
+  }
+]`, parsedURL.Host, parsedURL.Host)
+
+	if expectedResultsJSON != string(resultsJSON) {
+		t.Fatalf("results json does not match: %s vs %s", resultsJSON, expectedResultsJSON)
+	}
+}
+
+// config must be loaded from file paths, this creates a tmp file with the host
+// to load in for the DataGatherer
+func createDgHostConfigWithServer(server string) (*os.File, error) {
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a tmpfile for host")
+	}
+
+	if _, err := tmpfile.Write([]byte(server)); err != nil {
+		return nil, fmt.Errorf("failed to write to tmp host file")
+	}
+	if err := tmpfile.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close tmp host file after writing")
+	}
+
+	return tmpfile, nil
+}
+
+// creates a kubeconfig file on disk with a reference to the local server
+// allowing requests to be mocked
+func createKubeConfigWithServer(server string) (*os.File, error) {
```

Comment:
Think it makes sense to return just the filename here since the file is not longer open, and we only really need the name anyway.

Reply:
We need to delete the file at the end of the test case.

Reply:
Indeed, but it is safer to only return file.Name(), rather than the *os.File which has already been closed. Taking a minimal footprint too, we only need the name string, rather than the entire os.File struct.

Reply:
Ahh, my bad. I thought we needed that for os.Remove... 

Fixed in c354eb7a10592e5347082bbe9e13e49f4ca62255